### PR TITLE
Refactor backend by following internal audit comments

### PR DESCRIPTION
**Pull request by sifnoc**
Updates:
- Create bash script(fixed the revert issue) 
- Removed `address_hashes`
- Refactored `MstInclusionProof` struct
- Fix error propagation on the async functions with try operator
- Fix some assesrt code
- Removed function `get_deployment_address` 
- Updated README
	- [Build a Solvency Verifier Contract](https://github.com/summa-dev/summa-solvency/tree/master/zk_prover#build-a-solvency-verifier-contract): the verifier generated base on specific `ptau` file.
	- [Generating Verifiers for Backend](https://github.com/summa-dev/summa-solvency/blob/master/backend/README.md#generating-verifiers-for-backend): Introduce bash script instead of
- Checked a unix `timestamp` as `u64`

### CODE REVIEW

**Review by enricobottazzi — CHANGES_REQUESTED** [review]

File: backend/README.md [hunk 1]
```diff
@@ -48,7 +48,27 @@ SIGNATURE_VERIFICATION_MESSAGE="Summa proof of solvency for CryptoExchange" carg
 
 ### Generating Verifiers for Backend
```
[the remark this answers is not in the document]
Comment (sifnoc) [reply backend/README.md:49]: Good point! That's missing in here. I will update for that.

File: backend/README.md [hunk 2]
```diff
@@ -97,12 +114,16 @@ If executed successfully, you'll see:
 
 ### 2. Submit Proof of Solvency
 
-This step is crucial for two primary reasons: first, to validate the root hash of the Merkle Sum Tree (`mst_root`); and second, to ensure that the assets held by the CEX exceed their liabilities, as confirmed through the proof verification on the Summa contract.
-The CEX must submit this proof of solvency to the Summa contract. Currently, it's a mandatory requirement to provide this proof before generating the inclusion proof for each user in the current round.
+This step is also crucial for two primary reasons:
```
[the remark this answers is not in the document]
Comment (enricobottazzi) [reply backend/README.md:97]: Exactly!!

**Review by alxkzmn — COMMENTED** [review]
Why did you decide to remove `get_deployment_address`? Now you need to supply this address manually when creating the signer, and the `deployments` file generation is unnecessary.

**Comment by sifnoc** [comment]
> Why did you decide to remove `get_deployment_address`? Now you need to supply this address manually when creating the signer, and the `deployments` file generation is unnecessary.

Well, I found that the method `get_deployment_address` was not being used effectively in both the `tests` and `example/summa_solvency_flow`. It seemed more straightforward to retrieve the address directly from the `summa_contract` instance.
 While I can see its potential usefulness in real-world scenarios, it didn't seem necessary for current implementation.
If you have any other examples or scenarios in mind where it could be useful, I'd love to hear your thoughts.

**Review by enricobottazzi — APPROVED** [review]
LGTM ∑

**Comment by sifnoc** [comment]
@alxkzmn Now the Signer accept `Address` and also `Path` of file contains address as config.

**Comment by alxkzmn** [comment]
Awesome!

**Pull request merged**